src/graph/graph-ops.ts

```diff
@@ -49,63 +39,80 @@ export const changeBadge =
     })),
   })
 
-export const findNeighborNodes = (graph: Graph, id: string) => [
-  ...graph.edges.filter(edge => edge.source === id).map(edge => edge.target),
-  ...graph.edges.filter(edge => edge.target === id).map(edge => edge.source),
+export const highlightGraph = (edgesToHighlight: Edge[]) => (graph: Graph) => {
+  const allTargetsAndSources = edgesToHighlight.reduce(
+    (acc: string[], edge: Edge) => [...acc, edge.source, edge.target],
+    []
+  )
+  const nodeIds = [...new Set(allTargetsAndSources)]
+
+  return {
+    edges: graph.edges.map((edge: Edge) => {
+      if (!edgesToHighlight.find((hEdge: Edge) => hEdge.id === edge.id)) {
+        return edge
+      }
+
+      return {
+        ...edge,
+        style: { stroke: HIGHLIGHT_COLOR },
+      }
+    }),
+    nodes: graph.nodes.map((node: Node) => {
+      if (!nodeIds.includes(node.id)) {
+        return node
+      }
+
+      return {
+        ...node,
+        data: { ...node.data, highlight: !node.data.highlight },
+      }
+    }),
+  }
+}
+export const findNeighborEdges = (graph: Graph, id: string): Edge[] => [
+  ...graph.edges.filter(edge => edge.source === id),
+  ...graph.edges.filter(edge => edge.target === id),
 ]
 
-const childNodesHelper = (
+const findEdgeHelper = (
   edges: Edge[],
-  head: string,
+  head: Edge,
   queue: Edge[],
-  visited: string[] = []
-): string[] => {
+  visited: Edge[],
+  isUpstream: boolean = false
+): Edge[] => {
   if (!queue.length) {
     return visited
   }
 
-  const neighbors = edges.filter(edge => edge.source === head && !visited.includes(edge.target))
+  const cmp = (edge: Edge, head: Edge) =>
```

Comment:
what is cmp?

Reply:
It is sort of like compare. I should change that 😁

Reply:
changed to `matchEdges`